QtCollider/QcSignalSpy.h

```diff
@@ -91,7 +91,12 @@ class QcSignalSpy : public QObject {
 
             for (int i = 0; i < _argTypes.count(); ++i) {
                 QMetaType::Type type = static_cast<QMetaType::Type>(_argTypes.at(i));
-                args << QVariant(type, argData[i + 1]);
+                if (type == QMetaType::QVariant) {
+                    // avoid creating a QVariant<QVariant>
+                    args << QVariant(type, argData[i + 1]).value<QVariant>();
```

Comment:
i think this is exactly the same, just fewer steps :)

```suggestion
                    args << argData[i + 1]);
```

Reply:
This is also what I would think, but then it complains that QVariant(void*) is a deleted function.
```
QcSignalSpy.h:96:40: error: use of deleted function ‘QVariant::QVariant(void*)’
   96 |                     args << argData[i+1];

/usr/include/qt/QtCore/qvariant.h:499:12: note: declared here
  499 |     inline QVariant(void *) = delete;
```
Does this mean that Qt explicitly forbids "casting" a (void*) to QVariant?

Reply:
ohh i missed that was the type of argData (`void**`). yeah, this is fine the way it is, i think this would also work:

```suggestion
                    args << *static_cast<const QVariant*>(argData[i + 1]);
```

but that's just a hunch.

`inline QVariant(void *) = delete;` is a constructor for QVariant, when marked with "delete" it means the compiler will forbid calling it.